apps/openmw/mwmechanics/actors.cpp

```diff
@@ -1591,6 +1596,30 @@ namespace MWMechanics
                 getActorsSidingWith(*it, out);
     }
 
+    void Actors::getActorsSidingWith(const MWWorld::Ptr &actor, std::set<MWWorld::Ptr>& out, std::map<const MWWorld::Ptr, const std::set<MWWorld::Ptr> >& cachedAllies) {
+        std::list<MWWorld::Ptr> followers = getActorsSidingWith(actor);
+
+        // If we have already found actor's allies, use the cache
+        std::map<const MWWorld::Ptr, const std::set<MWWorld::Ptr> >::const_iterator search = cachedAllies.find(actor);
+        if (search != cachedAllies.end())
+            out = search->second;
+        else
+        {
+            for (std::list<MWWorld::Ptr>::iterator it = followers.begin(); it != followers.end(); ++it)
+                if (out.insert(*it).second)
+                    getActorsSidingWith(*it, out);
```

Comment:
I don't think it matters, because you shouldn't come across anything in the cache while iterating through getActorsSidingWith(*it, out) anyway, right? Again, sorry if I'm misunderstanding.